esrally/metrics.py

```diff
@@ -250,7 +251,22 @@ def results_template(self):
 
     def _read(self, template_name):
         with open("%s/resources/%s.json" % (self.script_dir, template_name), encoding="utf-8") as f:
-            return f.read()
+            # if we're using a persistent metrics store, then we should check and update the index templates settings
+            if self.cfg.opts("reporting", "datastore.type") == "elasticsearch":
+                return self._update_template(f)
+
+            else:
+                return f.read()
+
+    def _update_template(self, template_name):
+        template = json.load(template_name)
+        template["settings"]["index"]["number_of_shards"] = self.cfg.opts(
```

Comment:
somehow, referencing config parameters directly here feels (to me) slightly out of place.
I'd have defined them as instance variables instead e.g. `self._number_of_shards = self.cfg.opts("reporting", "datastore.number_of_shards", default_value=1, mandatory=False)` and `self._number_of_replicas = ...` for better readability and discoverability but obviously what you have here works fine too.